lib/SPIRV/SPIRVReader.cpp

```diff
@@ -3224,11 +3224,17 @@ bool SPIRVToLLVM::transAddressingModel() {
   switch (BM->getAddressingModel()) {
   case AddressingModelPhysical64:
     M->setTargetTriple(SPIR_TARGETTRIPLE64);
-    M->setDataLayout(SPIR_DATALAYOUT64);
+    M->setDataLayout(BM->hasProgramAddressSpace()
+                         ? std::string(SPIR_DATALAYOUT64) + "-P" +
+                               std::to_string(BM->getProgramAddressSpace())
+                         : SPIR_DATALAYOUT64);
```

Comment:
It would be nice to put this logic into a separate function that takes `BM` and `SPIR_DATALAYOUT64`. This way you will avoid the copy-paste below.